go/private/context.bzl

```diff
@@ -952,6 +998,7 @@ go_config = rule(
     doc = """Collects information about build settings in the current
     configuration. Rules may depend on this instead of depending on all
     the build settings directly.""",
+    toolchains = [GO_TOOLCHAIN],
```

Comment:
Adding a toolchain dependency here likely means that in case of a missing Go toolchain, Bazel will fail on this target (as a transitive dep of all Go targets) instead of an actual user-defined Go target. That's only marginally worse (or maybe even better?) than failing on some random actual `go_library` that's a leaf in the dep tree.

Could you check how the "missing toolchain" error changes with this, e.g. by targeting an exotic platform?

Reply:
It's the same as on master, it looks like the first failure at the top-level target is reported

```
platform(
    name = "exotic_platform",
    constraint_values = [
        "@platforms//os:haiku",
        "@platforms//cpu:mips64",
    ],
)
```

```
➜  rules_go git:(zbarsky/more-mode2) ✗ bazel build //tests/core/build_stdlib:build_stdlib_test  --platforms=//:exotic_platform                                                                                                                                                     ~/rules_go
ERROR: /Users/dzbarsky/rules_go/tests/core/build_stdlib/BUILD.bazel:3:14: While resolving toolchains for target //tests/core/build_stdlib:build_stdlib_test (dd044de): No matching toolchains found for types //go:toolchain.
To debug, rerun with --toolchain_resolution_debug='//go:toolchain'
If platforms or toolchains are a new concept for you, we'd encourage reading https://bazel.build/concepts/platforms-intro.
```

Reply:
`bazel build //tests/core/...  --platforms=//:exotic_platform` fails on a random test in that package when run a few times

Reply:
Nice, guess I still haven't fully understood how toolchain resolution works :-) Thanks for confirming this!